scripts/shared/cloud-prepare.sh

```diff
@@ -27,18 +27,55 @@ function check_gateway_exists() {
     [[ $(kubectl get nodes -l "${GATEWAY_LABEL}" --no-headers | wc -l) -gt 0 ]]
 }
 
-function prepare_kind() {
-    read -r -a nodes <<< "${cluster_nodes[$cluster]}"
-    local node=${cluster}-${nodes[-1]}
-    kubectl label node "$node" "$GATEWAY_LABEL" --overwrite
+function check_and_apply_air_gap_env() {
+    local node="$1"
+    local pub_ip
 
     if [[ "$AIR_GAPPED" = true ]]; then
-        local pub_ip
+        echo "Apply Air Gap environment configuration"
         pub_ip=$(kubectl get nodes "$node" -o jsonpath="{.status.addresses[0].address}")
         kubectl annotate node "$node" gateway.submariner.io/public-ip=ipv4:"$pub_ip"
     fi
 }
 
+function prepare_kind() {
+    read -r -a nodes <<< "${label_gw_nodes[$cluster]}"
+
+    worker_count=$(grep -o worker <<< "${nodes[*]}" | wc -l || true)
+    master_count=$(grep -o control-plane <<< "${nodes[*]}" | wc -l || true)
+
+    if [[ "$worker_count" -gt 0 ]]; then
+        echo "Label $worker_count worker nodes as gateway"
+        worker_nodes=$(kubectl get nodes --no-headers=true \
+            -o custom-columns=NAME:.metadata.name | grep worker | head -n "$worker_count")
+
+        for node in ${worker_nodes}; do
+            kubectl label node "${node}" "${GATEWAY_LABEL}" --overwrite
+
+            check_and_apply_air_gap_env "$node"
+        done
+    else
```

Comment:
This will just always label the last node in case you didnt specify any worker nodes to label.

Reply:
@mkolesnik 
Yes, that's the point.
It's a default config that the flow will fall back to if "label-gateway" parameter will not be specified.

Reply:
You're missing the point of my original comment, in case you specify `label-gateway: control-plane` it would still label the last node - and in case you have any worker nodes, you will likely end up with 2 gateways.

Reply:
Fixed.